Runigram.java

```diff
@@ -37,93 +41,143 @@ public static Color[][] read(String fileName) {
 		in.readInt();
 		// Creates the image array
 		Color[][] image = new Color[numRows][numCols];
-		// Reads the RGB values from the file, into the image array. 
-		// For each pixel (i,j), reads 3 values from the file,
-		// creates from the 3 colors a new Color object, and 
-		// makes pixel (i,j) refer to that object.
-		//// Replace the following statement with your code.
-		return null;
+		// Reads the RGB values from the file, into the image array.
+		for (int i = 0; i < image.length; i++) {
+			for (int j = 0; j < image[0].length; j++) {

```

Comment:
row and col is better variable names than i and j. Also instead of image.length you can use numCols and numRows that was extracted before.